source/numerics/data_out_dof_data.cc

```diff
@@ -547,7 +547,13 @@ namespace internal
     get_function_values (const FEValuesBase<DH::dimension,DH::space_dimension> &fe_patch_values,
                          std::vector<dealii::Vector<double> >    &patch_values_system) const
     {
-      fe_patch_values.get_function_values (*vector, patch_values_system);
+      //FIXME
+      std::vector<dealii::Vector<typename VectorType::value_type> > tmp(patch_values_system.size());
+      for (unsigned int i = 0; i < patch_values_system.size(); i++)
+        tmp[i].reinit(patch_values_system[i]);
```

Comment:
i would say the whole `DataOut` (plus auxiliary classes) should be reworked a lot to support number templating. And then in `.write_*` some magic should happen to output. Perhaps, with automatic appending of `_real` or `_imag` to the vector name (in case of VTK).